File: common/src/test/scala/com/kotori316/fluidtank/tank/TierTest.java
package com.kotori316.fluidtank.tank;

import org.junit.jupiter.api.DynamicTest;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestFactory;
import scala.math.BigInt;

import java.util.EnumMap;
import java.util.stream.Stream;

import static org.junit.jupiter.api.Assertions.*;

class TierTest {
    @TestFactory
    Stream<DynamicTest> testGetCapacity() {
        return DynamicTest.stream(Stream.of(Tier.values()), Tier::toString, t -> assertDoesNotThrow(t::getCapacity));
    }

    @Test
    void checkWoodCapacity() {
        assertEquals(BigInt.apply(81000 * 4), Tier.WOOD.getCapacity());
    }

    @Test
    void checkStoneCapacity() {
        assertEquals(BigInt.apply(81000 * 16), Tier.STONE.getCapacity());
    }

    @Test
    void setBadMap() {
        assertThrows(IllegalArgumentException.class, () -> Tier.setCapacityMap(new EnumMap<>(Tier.class)));
    }
}

File: common/src/test/scala/com/kotori316/fluidtank/tank/TierTest.scala
package com.kotori316.fluidtank.tank

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.{DynamicTest, Test, TestFactory}

class TierTest {
  @TestFactory
  def testGetCapacity: Array[DynamicTest] = {
    Tier.values
      .map(t => DynamicTest.dynamicTest(t.toString, () => assertDoesNotThrow(() => t.getCapacity)))
  }

  @Test
  def checkWoodCapacity(): Unit = {
    assertEquals(BigInt.apply(81000 * 4), Tier.WOOD.getCapacity)
  }

  @Test
  def checkStoneCapacity(): Unit = {
    assertEquals(BigInt.apply(81000 * 16), Tier.STONE.getCapacity)
  }

  @Test
  def setBadMap(): Unit = {
    assertThrows(classOf[IllegalArgumentException], () => Tier.setCapacityMap(new java.util.EnumMap[Tier, BigInt](classOf[Tier])))
  }
}